packages/dev/test-utils/src/dnd.ts

```diff
@@ -0,0 +1,114 @@
+import {act, fireEvent} from '@testing-library/react';
+// eslint-disable-next-line monorepo/no-internal-import
+import {DataTransfer, DragEvent} from '@react-aria/dnd/test/mocks';
+
+type DragPointerType = 'mouse' | 'touch' | 'pointer';
+
+type Delta = {x: number, y: number}
+
+function getElementCenter(element: Element) {
+  let {left, top, width, height} = element.getBoundingClientRect();
+  return {
+    x: left + width / 2,
+    y: top + height / 2
+  };
+}
+
+async function performDrag(source: Element, target: Element, delta: Delta, type: DragPointerType = 'mouse', steps = 1, duration = 0) {
+  let from = getElementCenter(source);
+  let to = {x: 0, y: 0};
+
+  if (target) {
+    to = getElementCenter(target);
+  } else if (delta) {
+    to = {x: from.x + delta.x, y: from.y + delta.y};
+  }
+
+
+  let step = {
+    x: (to.x - from.x) / steps,
+    y: (to.y - from.y) / steps
+  };
```

Comment:
if there are steps, it may be a good idea to fire a callback and allow people to insert their own code between steps. Maybe they are dragging over some element in the middle of the page on their way.

Reply:
Thats a good idea. I included placeholders for duration and steps, but wasn't sure how to go about the api yet.